README.md

```diff
@@ -31,6 +31,61 @@ Here is a brief list of the available widgets:
 * **edit_box**: an editable text box (multi-line edit_boxes are not yet fully supported).
 * **scroll_frame**: a frame that has scrollable content.
 
+# Getting started 
+first, clone the project
+```
+git clone https://github.com/cschreib/lxgui
+``` 
+
+install the dependencies for your operating system
+
+Ensure your c++ compiler is up to date. Lxgui requires a compiler that is c++17 compliant (GCC >= 8, clang >= 7, Apple-Clang >= 11, or VisualStudio >= 2017)
+
+
+- dnf based distros (fedora)
+```
+sudo dnf install -y SFML-devel lua-devel
+``` 
+
+- apt based distros (Debian, Ubuntu)
+``` 
+sudo apt install libsfml-dev liblua5.2-dev
+``` 
+- OSX 
+``` 
+ brew install sfml lua
+``` 
+- Windows 
+``` 
+ vcpkg install sfml lua
+```
+
+then initialize the submodules
+```
+cd lxgui
+git submodule update --init
+```
+
+# OpenGL setup 
+dnf based distros (fedora) 
+```
+sudo dnf install -y freetype-devel SFML-devel lua-bit32 mesa-libGLU-devel lua-devel
+```
+
+## install GLEW                                                                  
+download tar of the [![latest release](https://github.com/nigels-com/glew/releases/)](https://github.com/nigels-com/glew/releases/)                                                                                 
+```
+sudo dnf -y install libXmu-devel libXi-devel libGL-devel dos2unix git wget gcc
```

Comment:
If this stays, probably `dos2unix git wget gcc` shouldn't be here.

Reply:
oh lol my friend added that because he was building it in an empty buildah container. can remove!